packages/zosjobs/__tests__/__system__/DownloadJobs.system.test.ts

```diff
@@ -289,8 +290,86 @@ describe("Download Jobs - System tests", () => {
                 }
             }
         }, LONG_TIMEOUT);
-    });
 
+        it("should be able to download all DDs from job output with a record range (0-5)", async () => {
+            for (const file of jobFiles) {
+                if (file.ddname === "JESMSGLG") {
+                    jesJCLJobFile = file;
+                }
+            }
+
+            await DownloadJobs.downloadAllSpoolContentCommon(REAL_SESSION, {
+                outDir: outputDirectory,
+                jobid,
+                jobname,
+                recordRange: "0-5"
+            });
+            const expectedFile = DownloadJobs.getSpoolDownloadFilePath(
+                {
+                    jobFile: jesJCLJobFile,
+                    omitJobidDirectory: false,
+                    outDir: outputDirectory
+                }
+            );
+            expect(IO.existsSync(expectedFile)).toEqual(true);
+            expect(IO.readFileSync(expectedFile).toString()).toBeDefined();
+            expect(IO.readFileSync(expectedFile).toString()).toContain("J E S 2  J O B  L O G");
+            expect(IO.readFileSync(expectedFile).toString()).not.toContain("0------ JES2 JOB STATISTICS ------");
```

Comment:
In addition to testing for key entries within the spool, maybe you could test for the total number of lines that were retrieved. This would confirm that different ranges actually captured different amounts of data.